lib/spack/spack/cmd/context.py

```diff
@@ -0,0 +1,587 @@
+        help="The context you are working with"
+    )
+
+
+def setup_parser(subparser):
+    sp = subparser.add_subparsers(metavar='SUBCOMMAND', dest='context_command')
+
+    create_parser = sp.add_parser('create', help='Make a context')
+    add_common_args(create_parser)
+    create_parser.add_argument(
+        '--init-file', dest='init_file',
+        help='File with user specs to add and configuration yaml to use'
+    )
+
+    add_parser = sp.add_parser('add', help='Add a spec to a context')
+    add_common_args(add_parser)
+    add_parser.add_argument(
+        'package',
+        nargs=argparse.REMAINDER,
+        help="Spec of the package to add"
+    )
+
+    remove_parser = sp.add_parser(
+        'remove', help='Remove a spec from this context')
+    add_common_args(remove_parser)
+    remove_parser.add_argument(
+        'package',
+        nargs=argparse.REMAINDER,
+        help="Spec of the package to remove"
+    )
+
+    concretize_parser = sp.add_parser(
+        'concretize', help='Concretize user specs')
+    add_common_args(concretize_parser)
+
+    relocate_parser = sp.add_parser(
+        'relocate', help='Reconcretize context with new OS and/or compiler')
+    add_common_args(relocate_parser)
+    relocate_parser.add_argument(
+        '--compiler',
+        help="Compiler spec to use"
+    )
+
+    list_parser = sp.add_parser('list', help='List specs in a context')
+    list_parser.add_argument(
+        '--include-deps', action='store_true',
+        dest='include_deps', help='Show dependencies of requested packages')
+    add_common_args(list_parser)
+
+    modules_parser = sp.add_parser(
+        'list-modules',
+        help='Show modules for for packages installed in a context')
+    add_common_args(modules_parser)
+
+    upgrade_parser = sp.add_parser(
+        'upgrade',
+        help='Upgrade a dependency package in a context to the latest version')
+    add_common_args(upgrade_parser)
+    upgrade_parser.add_argument(
+        'dep_name', help='Dependency package to upgrade')
+    upgrade_parser.add_argument(
+        '--dry-run', action='store_true', dest='dry_run',
+        help="Just show the updates that would take place")
+
+    stage_parser = sp.add_parser(
+        'stage',
+        help='Download all source files for all packages in a context')
+    add_common_args(stage_parser)
+
+    config_update_parser = sp.add_parser(
+        'update-config',
+        help='Add config yaml file to context')
+    add_common_args(config_update_parser)
+    config_update_parser.add_argument(
+        'config_files',
+        nargs=argparse.REMAINDER,
+        help="Configuration files to add"
+    )
+
+    install_parser = sp.add_parser(
+        'install',
+        help='Install all concretized specs in a context')
+    add_common_args(install_parser)
+
+
+def context(parser, args, **kwargs):
+    action = {
+        'create': context_create,
+        'add': context_add,
+        'concretize': context_concretize,
```

Comment:
When you add specs with ```spack context add <context> <spec>``` it doesn't concretize them (which I just added to the latest doc commit), so the ```concretize``` command takes care of that. It does write to the context (but does not install the specs).